### ISSUE TITLE
Hombrew: review the code for the option "Qo"

### ISSUE THREAD

**Issue by icy** [icy/pacapt#23]
@Arcterus 

The code is as below

```
        _exec_ HOMEBREW "
            cd \"\$(dirname -- \$(which $_PKG))\"
            pkg=\"\$(pwd -P)/\$(basename -- $_PKG)\"
            prefix=\"\$(brew --prefix)\"
            cellar=\"\$(brew --cellar)\"
            for package in \$cellar/*; do
              files=(\${package}/*/\${pkg/#\$prefix\//})
              if [[ -e \${files[\${#files[@]} - 1]} ]]; then
                echo \"\${package/#\$cellar\//}\"
                break
              fi
            done
          "
```

The use of `which $_PKG` is very weird. It would be often a path to binary here, but $_PKG is not in our case. Please check and confirm if that line is really working.

Thank you.

**Comment by icy** [comment]
This code was added in the commit 2a668dc33d288967e6c1aad94a7905fb7ee884be . I've fixed in the commit e55063e9dde2371381f0a8361c3559231c6e14fc to prevent a shell expansion.

**Comment by Arcterus** [comment]
It worked before (FYI, it also works now).  Your commit only modified the location where the expansion occurs.  It doesn't really matter where, as the variables aren't being updated within the loop.

**Comment by Arcterus** [comment]
Also, I think that `which` should be `command -v` for the sake of consistency.

**Comment by icy** [comment]
On Tue, 08 Oct 2013 16:34:12 -0700
Arcterus notifications@github.com wrote:

> It worked before (FYI, it also works now).  Your commit only modified
> the location where the expansion occurs.

The early expansion will work on all system. For example, if `pacapt`
is executed on Debian system, the expansion happens, and generate
the error, because something is not available on Debian.

**Comment by Arcterus** [comment]
Ah, I see.  Well, in that case the only variables (from what I can tell) that NEED to be expanded latter are the ones involving `brew`, as that is the only Homebrew-specific command.  However, the new code is probably better.

**Comment by icy** [comment]
On Tue, 08 Oct 2013 19:48:48 -0700
Arcterus notifications@github.com wrote:

> Ah, I see.  Well, in that case the only variables (from what I can
> tell) that NEED to be expanded latter are the ones involving `brew`,
> as that is the only Homebrew-specific command. 

Yes that's right. The current script still has something wrong with
expansion, and I hope to fix them in #19 :)
